### ISSUE TITLE
Stream Data example raises an error

### ISSUE THREAD

**Issue by benwilson512** [elixir-lang/elixir-lang.github.com#1099]
Hey folks!

I'm beginning to look at the StreamData library finally, and discovered that the example given on the stream data blog here https://github.com/elixir-lang/elixir-lang.github.com/blob/master/_posts/2017-10-31-stream-data-property-based-testing-and-data-generation-for-elixir.markdown#using-property-based-testing-in-stream_data

```
    check all a <- integer(),
              b <- integer(),
              a > 0 and b > 0,
              sum = a + b do
      assert sum > a
      assert sum > b
    end
```

This gets 
```
** (StreamData.FilterTooNarrowError) too many consecutive elements were filtered out.
```

For me. If you write a pos_integer generator via
```
    pos_integer = StreamData.map(integer(), fn
      n when n > 0 -> n
      0 -> 1
      n -> n * -1
    end)
```

then it works fine.

Is the example wrong? Is the stream data library overly sensitive about filtering out values?

```
Erlang/OTP 20 [erts-9.2.1] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:10] [hipe] [kernel-poll:false]

Elixir 1.6.2 (compiled with OTP 19)
```
```
$ mix deps|grep stream_data
* stream_data 0.4.2 (Hex package) (mix)
  locked at 0.4.2 (stream_data) fa86b78c
```

**Comment by whatyouhide** [comment]
> Is the stream data library overly sensitive about filtering out values?

Well if it's _overly_ sensitive or not is up for discussion but this is a case of stream_data being sensitive. Since the initial generation size is very small (`1` to be precise), it's easy that sometimes either `a` or `b` are `0` or `-1` and the filter blows up. I'll find a better example with filtering and assignment.

**Comment by whatyouhide** [comment]
@benwilson512 closed in fa6630c094a6cd2ee1e7bfd9f59aa96b3a5c7be6.

**Comment by benwilson512** [comment]
Thanks!
